### PR TITLE
Update `ProphetModel` to handle external timestamp

### LINKED ISSUE #186: Update `ProphetModel` to handle integer timestamp
### 🚀 Feature Request

In #178 it was found that `ProphetModel` fails on data with integer timestamp. It could be fixed.

### Proposal

Fix tests `tests/test_models/test_inference/test_forecast/TestForecastOutSample` for models:
- `ProphetModel`

This model can't work with integer timestamp, it expects datetime. I haven't found a documentation page dedicated to learning on non-datetime data, there is a page about [learning on non-daily data](https://facebook.github.io/prophet/docs/non-daily_data.html).

Add parameter `timestamp_column`:
- Default value `None`
- Default behavior: model uses index for timestamp info
- Error if `timestamp_column` is `None` and integer timestamp is used
- Error if column isn't present in dataframe
- Error if given column doesn't consist of sequential timestamps with supported frequency (H, D)

### Test cases

- Update tests for `ProphetModel`
- Move fixed tests in `tests/test_models/test_inference/test_forecast/TestForecastOutSample` from fail to success category
- Move fixed tests `tests/test_models/test_inference/test_predict/TestPredictInSampleSuffix` from fail to success category

### Additional context

Blocked by #178.

### PR DESCRIPTION
**Pull request by d-a-bunin**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #186.

## Closing issues
Closes #186.

### CODE REVIEW

File: etna/models/prophet.py [hunk 1]
```diff
@@ -193,20 +203,45 @@ def predict(self, df: pd.DataFrame, prediction_interval: bool, quantiles: Sequen
...
+            if self.timestamp_column not in df.columns:
+                raise ValueError("Invalid timestamp_column! It isn't present in a given dataset.")
+
+            if self.timestamp_column not in self.regressor_columns:
+                raise ValueError("Invalid timestamp_column! It should be a regressor.")
+
+            if not pd.api.types.is_datetime64_dtype(df[self.timestamp_column]):
+                raise ValueError("Invalid timestamp_column! Only datetime type is supported.")
+
+            if len(df[self.timestamp_column]) >= 3 and pd.infer_freq(df[self.timestamp_column]) is None:
```
Comment (d-a-bunin) [inline etna/models/prophet.py:223]: It doesn't check that frequency is always the same. For example, it works fine if we have one frequency for train and for test. In theory, prophet can work fine even if there no regular frequency, but I'm not sure should we support this case or not.

Comment (d-a-bunin) [reply etna/models/prophet.py:223]: We could probably infer frequency during train, for example, and check if it is as expected.

Comment (ostreech1997) [reply etna/models/prophet.py:223]: I think we should check the freq and give a warning if the freq for the train is different from the freq for the test.

File: tests/test_models/test_inference/conftest.py
Comment (ostreech1997) [inline tests/test_models/test_inference/conftest.py:1]: Why we don't add this code to `tests/conftest.py` ?

Comment (d-a-bunin) [reply tests/test_models/test_inference/conftest.py:1]: For now, it seems specific for inference tests. I'll think about moving it higher.

File: etna/models/prophet.py (hunk 1 above)
Comment (d-a-bunin) [reply etna/models/prophet.py:223]: I dont like the idea of warning. It will be thrown in every per-segment model, which seems too much.

Comment (ostreech1997) [reply etna/models/prophet.py:223]: I think it is a bad idea to have different frequencies, so the only option is to fail in this situation.

File: tests/test_models/test_prophet.py [hunk 2]
```diff
@@ -15,14 +16,26 @@
 from tests.test_models.utils import assert_sampling_is_valid
 
 
+@pytest.fixture
```
Comment (ostreech1997) [inline tests/test_models/test_prophet.py:19]: Code duplication (like in `tests/test_models/test_inference/conftest.py)`

**Comment by martins0n** [comment]
@d-a-bunin What do you think if Prophet would pick timestamp implicitly? 
We can choose `timestamp` column by default. If `timestamp` column is of type `int`, we should choose the next timestamp typed column. If there are multiple timestamp typed columns - choose the first one.

And we should specify this implicit logic

**Comment by d-a-bunin** [comment]
Where do you think this could be useful? I can see it useful for working with pre-defined models on different datasets, e.g. in auto-ml there set of configs is pre-defined.

- For now, it seems implicit, and I don't really like it. We could probably add such behavior in the future with some special parameter.
- It still breaks if we have multiple timestamp columns.
- We don't have such logic for `DateFlagsTransform`, `TimeFlagsTransform` and `HolidayTransform`.
- I don't know how should we make auto-ml configs work in all those conditions.

We could make smth simple (like it works now) and easy-to-improve and then improve it according to our needs later.

**Comment by martins0n** [comment]
> * It still breaks if we have multiple timestamp columns.

No, I we choose random one

> Where do you think this could be useful?

It's just more simple to support.

You've changed a lot of code here. Do we really want to change so many places just for supporting corner case?

**Comment by d-a-bunin** [comment]
> No, if we choose random one

It works unpredictable, I don't think it is a good idea. Moreover, you suggested selecting the first one in the first message.

> It's just simpler to support.

I don't think so. We have to make the same code changes + add logic for automatic column detection, which selects `timestamp_column` automatically instead of manually. I think that if we are going to provide some automatic selection we should also provide ability of a manual selection. 

Current solution could be extended into automatic in the future by adding, e.g. parameter `timestamp_mode` with two possible values: "auto", "manual" with default "manual" to save current behavior.

**Review by ostreech1997 — APPROVED** [review]

**Pull request merged**